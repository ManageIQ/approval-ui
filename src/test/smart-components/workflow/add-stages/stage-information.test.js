import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import StageInformation from '../../../../smart-components/workflow/add-stages/stage-information';

describe('<StageInformation />', () => {
  let initialProps;

  beforeEach(() => {
    initialProps = {
      formData: {
        name: 'Stage info name test',
        description: 'Stage info description test'
      },
      handleChange: jest.fn(),
      title: 'Set stage info test'
    };
  });

  it('should render correctly', () => {
    const wrapper = mount(<StageInformation { ...initialProps } />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
