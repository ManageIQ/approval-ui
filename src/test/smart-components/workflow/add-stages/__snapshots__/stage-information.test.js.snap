// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StageInformation /> should render correctly 1`] = `
<StageInformation
  formData={
    Object {
      "description": "Stage info description test",
      "name": "Stage info name test",
    }
  }
  handleChange={[MockFunction]}
  title="Set stage info test"
>
  <Stack
    gutter="md"
  >
    <div
      className="pf-l-stack pf-m-gutter"
    >
      <StackItem>
        <div
          className="pf-l-stack__item"
        >
          <Title
            size="md"
          >
            <h1
              className="pf-c-title pf-m-md"
            >
               
              Set stage info test
               
            </h1>
          </Title>
        </div>
      </StackItem>
      <StackItem>
        <div
          className="pf-l-stack__item"
        >
          <Form>
            <form
              className="pf-c-form"
              noValidate={true}
            >
              <FormGroup
                fieldId="workflow-name"
                isRequired={true}
                label="Workflow name"
              >
                <div
                  className="pf-c-form__group"
                >
                  <label
                    className="pf-c-form__label"
                    htmlFor="workflow-name"
                  >
                    <span
                      className="pf-c-form__label-text"
                    >
                      Workflow name
                    </span>
                    <span
                      aria-hidden="true"
                      className="pf-c-form__label-required"
                    >
                      *
                    </span>
                  </label>
                  <TextInput
                    aria-describedby="workflow-name"
                    aria-label={null}
                    className=""
                    id="workflow-name"
                    isDisabled={false}
                    isReadOnly={false}
                    isRequired={true}
                    isValid={true}
                    name="workflow-name"
                    onChange={[Function]}
                    type="text"
                    value="Stage info name test"
                  >
                    <input
                      aria-describedby="workflow-name"
                      aria-invalid={false}
                      aria-label={null}
                      className="pf-c-form-control"
                      disabled={false}
                      id="workflow-name"
                      name="workflow-name"
                      onChange={[Function]}
                      readOnly={false}
                      required={true}
                      type="text"
                      value="Stage info name test"
                    />
                  </TextInput>
                </div>
              </FormGroup>
              <FormGroup
                fieldId="workflow-description"
                label="Description"
              >
                <div
                  className="pf-c-form__group"
                >
                  <label
                    className="pf-c-form__label"
                    htmlFor="workflow-description"
                  >
                    <span
                      className="pf-c-form__label-text"
                    >
                      Description
                    </span>
                  </label>
                  <TextArea
                    aria-label={null}
                    className=""
                    id="workflow-description"
                    isRequired={false}
                    isValid={true}
                    name="workflow-description"
                    onChange={[Function]}
                    resizeOrientation="both"
                    type="text"
                    value="Stage info description test"
                  >
                    <textarea
                      aria-invalid={false}
                      aria-label={null}
                      className="pf-c-form-control"
                      id="workflow-description"
                      name="workflow-description"
                      onChange={[Function]}
                      required={false}
                      type="text"
                      value="Stage info description test"
                    />
                  </TextArea>
                </div>
              </FormGroup>
            </form>
          </Form>
        </div>
      </StackItem>
    </div>
  </Stack>
</StageInformation>
`;
